src/harness/testcases/WINNF_FT_S_MCP_testcase.py

```diff
@@ -0,0 +1,295 @@
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import hashlib
+import json
+import os
+import sas
+import sas_testcase
+from sas_test_harness import SasTestHarnessServer, generateCbsdRecords, \
+    generatePpaRecords
+from util import winnforum_testcase, configurable_testcase, writeConfig, \
+  loadConfig, makePpaAndPalRecordsConsistent 
+
+class MultiConstraintProtectionTestcase(sas_testcase.SasTestCase):
+
+  def setUp(self):
+    self._sas, self._sas_admin = sas.GetTestingSas()
+    self._sas_admin.Reset()
+
+  def tearDown(self):
+    pass
+
+  def generate_MCP_1_default_config(self, filename):
+    """ Generates the WinnForum configuration for MCP.1. """
+    # Load devices for SAS UUT for multiple iterations through multiple domain proxy's
+    device_1 = json.load(
+      open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    device_2 = json.load(
+      open(os.path.join('testcases', 'testdata', 'device_b.json')))
+    device_3 = json.load(
+      open(os.path.join('testcases', 'testdata', 'device_c.json')))
+    device_4 = json.load(
+      open(os.path.join('testcases', 'testdata', 'device_d.json')))
+    device_5 = json.load(
+      open(os.path.join('testcases', 'testdata', 'device_e.json')))
+    device_6 = json.load(
+      open(os.path.join('testcases', 'testdata', 'device_f.json')))
+    device_7 = json.load(
+      open(os.path.join('testcases', 'testdata', 'device_g.json')))
+    device_8 = json.load(
+      open(os.path.join('testcases', 'testdata', 'device_h.json')))	  
+
+    # Load Grant requests
+    grant_request_1 = json.load(
+      open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_request_2 = json.load(
+      open(os.path.join('testcases', 'testdata', 'grant_1.json')))
+    grant_request_3 = json.load(
+      open(os.path.join('testcases', 'testdata', 'grant_2.json')))
+    grant_request_4 = json.load(
+      open(os.path.join('testcases', 'testdata', 'grant_3.json')))
+    grant_request_5 = json.load(
+      open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_request_6 = json.load(
+      open(os.path.join('testcases', 'testdata', 'grant_1.json')))
+    grant_request_7 = json.load(
+      open(os.path.join('testcases', 'testdata', 'grant_2.json')))
+    grant_request_8 = json.load(
+      open(os.path.join('testcases', 'testdata', 'grant_3.json')))	  
+  
+    # device_b is Category B
+    # Load Conditional Data
+    self.assertEqual(device_2['cbsdCategory'], 'B') 
+    conditionals_device_2 = {
+        'cbsdCategory': device_2['cbsdCategory'],
+        'fccId': device_2['fccId'],
+        'cbsdSerialNumber': device_2['cbsdSerialNumber'],
+        'airInterface': device_2['airInterface'],
+        'installationParam': device_2['installationParam'],
+        'measCapability': device_2['measCapability']
+    }
```

Comment:
device_4 is also category B are we not preloading those conditionals?

Reply:
added for device 4.